lib/datasource/index.ts

```diff
@@ -55,25 +59,139 @@ const datasources: Record<string, Datasource> = {
   terraform,
 };
 
-const cacheNamespace = 'datasource-releases';
+export function baseUrlLegacyMassager(sourceUrl) {
```

Comment:
Is there any way we could extract these out into a separate folder, with individual files for each git host? Then just expose a method that calls the correct URL extractor?

This file is already overloaded with responsibility, and extracting this would make it easier to add hosts e.g. Azure DevOps

Reply:
It would be possible, but I would prefer it if we could clean up the code code and get it merge-ready before introducing yet more structural changes.

Would that be alright with you?